# Classification
Some of the most common supervised learnings tasks are regression and classification. In this notebook we will focus on classification. We will be using the MNIST dataset. It is a set of sabout 70,000 images of handwritten digits. Each image contains a corresponding label representing the actual digit.

Thankfully, Scikit-Learn contains a lot of functions that can be used to download datasets, with the MNIST dataset being one of them. You can download it like so:

In [1]:
from sklearn.datasets import fetch_openml
mnist = fetch_openml('mnist_784', version=1)
mnist.keys()

C:\Users\eriks\ml\env\lib\site-packages\sklearn\datasets\_openml.py:932: FutureWarning: The default value of `parser` will change from `'liac-arff'` to `'auto'` in 1.4. You can set `parser='auto'` to silence this warning. Therefore, an `ImportError` will be raised from 1.4 if the dataset is dense and pandas is not installed. Note that the pandas parser may return different data types. See the Notes Section in fetch_openml's API doc for details.
  warn(


dict_keys(['data', 'target', 'frame', 'categories', 'feature_names', 'target_names', 'DESCR', 'details', 'url'])

The structure of datasets loaded in by Scikit-Learn generally contain this:

- A `DESCR` key descrbing the dataset
- A `data` key with an array (one row per instance, one column per feature)
- A `target` key with an array of labels

Such as:

In [18]:
X, y = mnist['data'], mnist['target']
print(X.shape)
print(y.shape)
type(X)

(70000, 784)
(70000,)


pandas.core.frame.DataFrame

There are 70,000 images, and each image has 784 features. Also, there are 70,000 labels corresponding to each image. The reason each image has 784 features is because each image is 28x28 pixels. Each feature representes the pixe's white/black value. This range is repsresented from 0-255 (white-black). We can take a look at on digit from the dataset like so:

In [23]:
import matplotlib as mpl
import matplotlib.pyplot as plt

example_digit = X[0]
example_digit_image = example_digit.reshape(28, 28)

plt.imshow(example_digit_image, cmap='binary')
plt.axis('off')
plt.show()

KeyError: 0

In [10]:
y[0]

'5'

In [14]:
len(X)

70000